File: php/test_primes.php
<?php
/**
 * Ganesh Bhat
 * https://github.com/ganeshkbhat/fastprimecalculations
 * Fastest method of Prime Number Calculations
 * Comparision results for correctness Tests for Prime Numbers
*/

require_once("./primes.php");
require_once("./primes_alternate_ways.php");
$iterations = 100000;

function strbool($value){
    return $value ? 'true' : 'false';
}


$result = isPrimeConventionalWay(83);
echo "Is Prime 83 (isPrimeConventionalWay): ", strbool($result), "\n";
$result = isPrimeSquarerootWay(83);
echo "Is Prime 83 (prime): ", strbool($result), "\n";
$result = isPrimeSquarerootWay(83);
echo "Is Prime 83 (isPrimeSquarerootWay): ", strbool($result), "\n";
$result = prime(83);
echo "Is Prime 83 (prime): ", strbool($result), "\n";


$result = isPrimeConventionalWay(169);
echo "Is Prime 169 (isPrimeConventionalWay): ", strbool($result), "\n";
$result = isPrimeSquarerootWay(169);
echo "Is Prime 169 (prime): ", strbool($result), "\n";
$result = isPrimeSquarerootWay(169);
echo "Is Prime 169 (isPrimeSquarerootWay): ", strbool($result), "\n";
$result = prime(169);
echo "Is Prime 169 (prime): ", strbool($result), "\n";


$result = isPrimeConventionalWay(100007);
echo "Is Prime 100007 (isPrimeConventionalWay): ", strbool($result), "\n";
$result = isPrimeSquarerootWay(100007);
echo "Is Prime 100007 (prime): ", strbool($result), "\n";
$result = isPrimeSquarerootWay(100007);
echo "Is Prime 100007 (isPrimeSquarerootWay): ", strbool($result), "\n";
$result = prime(100007);
echo "Is Prime 100007 (prime): ", strbool($result), "\n";

$result = isPrimeConventionalWay(300530164787);
echo "Is Prime 300530164787 (isPrimeConventionalWay): ", strbool($result), "\n";
$result = isPrimeSquarerootWay(300530164787);
echo "Is Prime 300530164787 (prime): ", strbool($result), "\n";
$result = isPrimeSquarerootWay(300530164787);
echo "Is Prime 300530164787 (isPrimeSquarerootWay): ", strbool($result), "\n";
$result = prime(300530164787);
echo "Is Prime 300530164787 (prime): ", strbool($result), "\n";


function test_primecalculations() {
    global $iterations;
    $count = 0;
    $arr = [];
    for ($i = 1; $i <= $iterations; $i++) {
        $traditional = isPrimeConventionalWay($i);
        $newer = prime($i);
        if ($traditional == $newer) {
            $count += 1;
        } else {
            array_push($arr, array($traditional, $newer, $i));
        }
    }
    print("[count, iterations, Error Array] list: \n");
    var_dump($count, $iterations, $arr);
    if ($count === $iterations) {
        return true;
    }
    return false;
}

$result = test_primecalculations();
echo "Results for comparision of calculations: ", strbool($result);

File: python/test_primes.py
##
# 
# Ganesh Bhat
# https://github.com/ganeshkbhat/fastprimecalculations
# Fastest method of Prime Number Calculations
# Functions and Alternate ways for calculation of Prime Numbers
# 
# ** prime(n) ** is the Recommended Way for calculation checks of prime numbers
# All others are alternative ways
#
##

from primes import *
from primes_alternate_ways import *
import time
import math


print("Is Prime 83 isPrimeConventionalWay: ", isPrimeConventionalWay(83))
print("Is Prime 83 isPrimeSquarerootWay: ", isPrimeSquarerootWay(83))
print("Is Prime 83 prime (SUGGESTED): ", prime(83))
print("Is Prime 83 isprimeAKSWay: ", isprimeAKSWay(83))


print("Is Prime 169 isPrimeConventionalWay: ", isPrimeConventionalWay(169))
print("Is Prime 169 isPrimeSquarerootWay: ", isPrimeSquarerootWay(169))
print("Is Prime 169 prime (SUGGESTED): ", prime(169))
print("Is Prime 169 isprimeAKSWay: ", isprimeAKSWay(169))


print("Is Prime 100007 isPrimeConventionalWay: ", isPrimeConventionalWay(100007))
print("Is Prime 100007 isPrimeSquarerootWay: ", isPrimeSquarerootWay(100007))
print("Is Prime 100007 prime (SUGGESTED): ", prime(100007))
print("Is Prime 100007 isprimeAKSWay: ", isprimeAKSWay(100007))


print("Is Prime 300530164787 isPrimeConventionalWay: ", isPrimeConventionalWay(300530164787))
print("Is Prime 300530164787 isPrimeSquarerootWay: ", isPrimeSquarerootWay(300530164787))
print("Is Prime 300530164787 prime (SUGGESTED): ", prime(300530164787))
print("Is Prime 300530164787 isprimeAKSWay: ", isprimeAKSWay(300530164787))


def test_primecalculations():
    count = 0
    iterations = 100000
    arr = []
    for i in range(1, (iterations+1)):
        traditional = isPrimeConventionalWay(i)
        newer = prime(i)
        if (traditional == newer):
            count = count + 1
        else:
            arr.append([traditional, newer, i])
    print("[count, iterations, Error Array] list: ", count, iterations, arr)
    # print("[count, iterations, Error Items] list: ", count, iterations, len(arr))
    if (count == iterations):
        return True
    return False


print("Tests Passed: ", test_primecalculations())

